File: Arrays_and_Strings/python/2.py
def isPermuation(word1: str, word2: str):
    if len(word1) != len(word2):
        return False
    charMap = dict()
    for i in range(len(word1)):
        char1 = word1[i]
        char2 = word2[i]

        if isinstance(charMap.get(char1), int):
            charMap[word1[i]] += 1
            if charMap[word1[i]] == 0:
                charMap.pop(word1[i])

        else:
            charMap[word1[i]] = 1

        if isinstance(charMap.get(char2), int):
            charMap[word2[i]] -= 1
            if charMap[word2[i]] == 0:
                charMap.pop(word2[i])
        else:
            charMap[word2[i]] = -1

    return not bool(charMap)


print(isPermuation("hello", "helol"))
print(isPermuation("tello", "helol"))
print(isPermuation("hellp", "helpl"))
print(isPermuation("hello", "helol"))
print(isPermuation("hello", "heloL"))

File: Arrays_and_Strings/javascript/2.js

function isPermuation(word1, word2) {
    if(word1.length != word2.length){
        return false
    }
    let map = new Set()

    for(let i=0; i< word1.length; i++){
        char = word1[i]
        char2 = word2[i]

        if(!map.has(char)){
            map.add(char)
        } else {
            map.delete(char)
        }

        if(!map.has(char2)){
            map.add(char2)
        } else {
            map.delete(char2)
        }
    }
    return map.size == 0;
}

console.log(isPermuation("hello","helol"))
console.log(isPermuation("tello","helol"))
console.log(isPermuation("hellp","helpl"))
console.log(isPermuation("hello","helol"))
console.log(isPermuation("hello","heloL"))

